### ISSUE TITLE
jack plugin freezes mpv.

### ISSUE THREAD

**Issue by crocket** [alsa-project/alsa-plugins#1]
I'm using alsa-plugin-1.1.8

`mpv --audio-device=alsa/plug:rawjack /path/to/musicfile` freezes mpv.
mpv doesn't freeze only when it plays a wav file.
I have to press Ctrl+C twice to kill mpv.

**Comment by aqxa1** [comment]
I've experienced that problem too, but turned out it was a [faulty .asoundrc](https://wiki.archlinux.org/index.php/JACK_Audio_Connection_Kit#Playing_nice_with_ALSA). Try removing .asoundrc, as there should be a default Jack pcm, as of alsa-plugin-1.1.7.

In your example, you could use:

`mpv --audio-device=alsa/jack /path/to/musicfile`

You can also configure your .asoundrc to your liking, and point to the built in "jack" pcm.

**Comment by crocket** [comment]
```
cannot connect alsa-jack.jackP.17825.0:out_000 to alsa_pcm:playback_1
```

**Comment by trebmuh** [comment]
@crocket : did you change` /path/to/musicfile` to fit with your own configuration?

**Comment by crocket** [comment]
Yes, I made it play a youtube video.

**Comment by aqxa1** [comment]
@crocket Try putting the following in ~/.asoundrc:

```
pcm.jacktest {                                                                                                                 
        type jack                                                                                                              
        playback_ports {                                                                                                       
                0 system:playback_1                                                                                            
                1 system:playback_2                                                                                            
        }                                                                                                                      
        capture_ports {                                                                                                        
                0 system:capture_1                                                                                             
                1 system:capture_2                                                                                             
        }                                                                                                                      
        hint {                                                                                                                 
                show {                                                                                                         
                        @func refer                                                                                            
                        name defaults.namehint.basic                                                                           
                }                                                                                                              
                description "JACK Audio Connection Kit"                                                                        
        }                                                                                                                      
}
```

Then:
>mpv --ao=alsa --audio-device=alsa/jacktest <file>

If that works, check that /usr/share/alsa/alsa.conf.d/50-jack.conf exists, and has the same content. I suspect you have "alsa_pcm:" ports instead of "system:" (going by your error above) which causes the freezing issues for me. Those ports are also used in older examples of using the JACK plugin, so you might have an outdated file somewhere. Check /etc/alsa/conf.d as well, if it exists.

Also, you should be able to just use mpv's JACK output directly, without going through ALSA.

Like this:
>mpv --ao=jack <file>

**Comment by crocket** [comment]
It seems `--audio-device=alsa/plug:jacktest` freezes mpv, but `--audio-device=alsa/jacktest` doesn't.
However, when mpv exits, I see the following error messages.

```
JackGraphManager::Disconnect not connected port_src = 7 port_dst = 1
Unknown source port in attempted (dis)connection src_name [alsa-jack.jacktestP.15122.0:out_001] dst_name [system:playback_2]
```

**Comment by aqxa1** [comment]
It looks to have been fixed in alsa-plugins-1.1.9, which explains why I wasn't having the issue with the built in jack plugin. See: 613ac0c3483fb9bc0d010e8ed4ad50191500d641

**Comment by crocket** [comment]
I'm going to close it for now. I will reopen a new issue when the issue happens again.
